### PR TITLE
Implemented codegen for CTL record fix changes

### PR DESCRIPTION
**Pull request by gnumonik**
This branch contains the codegen tools needed for the CTL record fix updates (WIP)

### CODE REVIEW

**Comment by bladyjoker** [comment]
@gnumonik please fix the build and notify us when it's done so we can review this.

**Review by bladyjoker — CHANGES_REQUESTED** [review]
Please add tests where you compare against Haskell BuiltinData reps. Make sure to test `newtypes` and `records`

**Comment by gnumonik** [comment]
@bladyjoker need your help fixing the IPC so I don't have to literally run all the tests by hand. 

there's a lingering JSON issue with newtype records that we need to fix. 

but the stupid manual tests pass and that means this PR is just about done

File: test/RoundTrip/app/src/Main.purs [hunk 1]
```diff
@@ -34,6 +36,6 @@ main = do
           log ""
         Right testData -> do
           error ""
-          log $ stringify $ encodeJson testData
+          log $ show (toData testData)
```
Comment (bladyjoker) [inline test/RoundTrip/app/src/Main.purs]: FIXME: THe json roundtrip test hangs here

**Comment by bladyjoker** [comment]
@gnumonik 

```
Round trip tests (Purescript <-> Haskell)
  Round trip prerequisite tests
    `[test/RoundTrip/app] $ spago build` should work [✔]
    `[test/RoundTrip/app] $ spago build` should not warn of unused packages buildable [✔]
  with defaultBridge
    should have a Purescript process running [✔]
    should produce Aeson-compatible representations [✔]    
      +++ OK, passed 100 tests.
  with ledger bridge
    should have a Purescript process running [✔]
    should produce PlutusData compatible representations [✔]    
      +++ OK, passed 100 tests.

Finished in 2.9049 seconds
6 examples, 0 failures
```

**Comment by bladyjoker** [comment]
TODO:
- [ ] Resolve Argonaut vs PlutusTx integer conflict (Argonaut doesn't handle Integers, and PlutusTx doesn't handle Ints)
   - **STATUS**: Vladimir is trying to make CTL use Aeson solely so we can print into that?
- [ ] Specify Arbitrary for PlutusTx.AssocMap
- [ ] Embed plutus-ledger-api types into testing types

**Review by bladyjoker — APPROVED** [review]

**Comment by bladyjoker** [comment]
CI build is still broken, keeping it that way until I find a solution

**Pull request merged**